src/java.base/share/classes/java/time/temporal/TemporalAmount.java

```diff
@@ -96,6 +98,7 @@
  *
  * @since 1.8
  */
+@AnnotatedFor({"nullness"})
 public interface TemporalAmount {
```

Comment:
The docs for `get` and `getUnits` are not explicit about whether their `TemporalUnit` instances can be null.

Both JDK implementation of `TemporalAmount` reject nulls from `get` ([`Period`](https://docs.oracle.com/javase/8/docs/api/java/time/Period.html#get-java.time.temporal.TemporalUnit-), [`Duration`](https://docs.oracle.com/javase/8/docs/api/java/time/Duration.html#get-java.time.temporal.TemporalUnit-)) -- albeit by throwing `UnsupportedTemporalTypeException` instead of `NullPointerException`. This means that we need to omit `@Nullable` here if we want to be safe.

For `getUnits`, the `TemporalUnit` instances are an output, so we could in principle still have `@Nullable` there. But `java.time` is consistent in treating a null `TemporalUnit` as bogus: See, for example, [`Temporal.isSupported`](https://docs.oracle.com/javase/8/docs/api/java/time/temporal/Temporal.html#isSupported-java.time.temporal.TemporalUnit-), or see the use of similar `TemporalField` in [`TemporalAccessor.get`](https://docs.oracle.com/javase/8/docs/api/java/time/temporal/TemporalAccessor.html#get-java.time.temporal.TemporalField-). That probably argues for doing the same. (Alternatively, it argues that this one exception could have been intentional :) But I think that would be at least a little perverse.)